src/components/screens/SearchView.vue

```diff
@@ -80,99 +103,31 @@ export default {
       return debounce(() => this.search(), '400ms')
     },
     targetNameFilterOptions: function() {
-      if (this.publishedScoreSets.length > 0) {
-        const values = this.publishedScoreSets.map((s) => s.targetGenes.map((t) => _.get(t, 'name')))
-        const valueFrequencies = _.countBy(values)
-        return _.sortBy(_.keys(valueFrequencies)).map((value) => ({value, badge: valueFrequencies[value]}))
-      } else {
-        return null
-      }
+      return countTargetGeneMetadata(this.publishedScoreSets, (targetGene) => targetGene.name)
     },
     targetOrganismFilterOptions: function() {
-      if (this.publishedScoreSets.length > 0) {
-        const values = this.publishedScoreSets.map((s) => _.concat(s.targetGenes.map((t) => _.get(t, 'targetSequence.reference.organismName', 'Accession Based Scoresets'))).flat()).flat()
-        const valueFrequencies = _.countBy(values)
-        return _.sortBy(_.keys(valueFrequencies)).map((value) => ({value, badge: valueFrequencies[value]}))
-      } else {
-        return null
-      }
+      return countTargetGeneMetadata(this.publishedScoreSets, 
+        (targetGene) => targetGene.targetSequence?.reference.organismName || 'Accession Based Scoresets')
     },
     targetAccessionFilterOptions: function() {
-      if (this.publishedScoreSets.length > 0) {
-        const values = this.publishedScoreSets.map((s) => _.concat(s.targetGenes.map((t) => _.get(t, 'targetAccession.accession', 'Sequence Based Scoresets'))).flat()).flat()
-        const valueFrequencies = _.countBy(values)
-        return _.sortBy(_.keys(valueFrequencies)).map((value) => ({value, badge: valueFrequencies[value]}))
-      } else {
-        return null
-      }
+      return countTargetGeneMetadata(this.publishedScoreSets,
+        (targetGene) => targetGene.targetAccession?.accession || 'Sequence Based Scoresets')
```

Comment:
This was a problem with the original logic that I am only noticing now, but creating this `Sequence Based Scoresets` (and similarly the `Accession Based Scoresets`) option will end up sending a non-sensical filter request to the API:
```
{"text":null,"targets":null,"targetOrganismNames":null,"targetAccessions":["Sequence Based Scoresets"],"targetTypes":null,"authors":null,"databases":null,"journals":null}
```

It might not make sense to handle in a refactor branch, especially because it may require API changes as well, but we should consider sending a different payload than the filter label for this sort of an option. Alternatively, maybe it makes more sense to not include a filter option for sequence based stuff in the accession filter menu at all.

Reply:
For now, let's not display them in the list. We can potentially work out something different in a separate issue/PR (but I think I actually like them not being represented in the filter list here).